api/types/types.go

```diff
@@ -87,12 +87,38 @@ type DiskUsageOptions struct {
 // DiskUsage contains response of Engine API:
 // GET "/system/df"
 type DiskUsage struct {
-	LayersSize  int64
-	Images      []*image.Summary
-	Containers  []*container.Summary
-	Volumes     []*volume.Volume
-	BuildCache  []*BuildCache
-	BuilderSize int64 `json:",omitempty"` // Deprecated: deprecated in API 1.38, and no longer used since API 1.40.
+	Images     *ImageDiskUsage
+	Containers *ContainerDiskUsage
+	Volumes    *VolumeDiskUsage
+	BuildCache *BuildCacheDiskUsage
+}
+
+// ImageDiskUsage contains the image disk usage information.
+type ImageDiskUsage struct {
+	TotalSize   int64
```

Comment:
I probably need to checkout this branch locally, but would it be possible to put the types in the relevant sub-packages (containers, images, ...)

We've been trying to scope these types to relevant types, because the top-level `types/` packages became a grab-bag of everything, also enforcing (indirect) imports in places where it was less than desirable.

We could have those packages imported in `system/` for use in `docker system du` , but also consider that we may have similar commands at some point for individual object-types (`docker container du` or similar), to go along with `docker container prune`.

Reply:
Doing this actually made things substantially easier. I've moved these types.